Mlem/API/Requests/CreateCommentLikeRequest.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  CreateCommentLikeRequest.swift
+//  Mlem
+//
+//  Created by Nicholas Lawson on 09/06/2023.
+//
+
+import Foundation
+
+struct CreateCommentLikeRequest: APIRequest {
+    
+    typealias Response = CreateCommentLikeResponse
+    
+    let endpoint: URL
+    let method: HTTPMethod
+    
+    struct Body: Encodable {
+        let auth: String
+        let comment_id: Int
+        let score: Int
+    }
+    
+    init(
+        account: SavedAccount,
+        commentId: Int,
+        score: ScoringOperation
+    ) throws {
+        do {
+            let data = try JSONEncoder().encode(
+                Body(
+                    auth: account.accessToken,
+                    comment_id: commentId,
+                    score: score.rawValue
+                )
+            )
```

Comment:
I'm not entirely sure if I like the boilerplate of having to put the encoder instantiation and .encode function inside every .init every time, plus having to wrap everything in a try-catch, couldn't we just chuck a Body at an extension function or something, and have that deal with the encoding issue?

Reply:
makes sense - I'll take a look at that 👍

Reply:
addressed by dbfe9740835ee1db247427226641e3617925b840